definitions/features/foreman_proxy.rb

```diff
@@ -6,30 +6,34 @@ class Features::ForemanProxy < ForemanMaintain::Feature
     end
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  FOREMAN_PROXY_SETTINGS_PATHS = ['/etc/foreman-proxy/settings.yml',
+                                  '/usr/local/etc/foreman-proxy/settings.yml'].freeze
+  DHCPD_CONFIG_PATHS = ['/etc/dhcp3/dhcpd.conf',
+                        '/etc/dhcp/dhcpd.conf',
+                        '/usr/local/etc/dhcpd.conf'].freeze
+
+  attr_reader :dhcpd_conf_file, :settings_file, :proxy_settings
 
   def initialize
-    @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
-    @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+    @dhcpd_conf_file = lookup_into(DHCPD_CONFIG_PATHS)
+    @settings_file = lookup_into(FOREMAN_PROXY_SETTINGS_PATHS)
+    @proxy_settings || load_proxy_settings
   end
 
   def valid_dhcp_configs?
     dhcp_req_pass? && !syntax_error_exists?
   end
 
-  def services
-    {
-      'foreman-proxy'            => 20,
-      'smart_proxy_dynflow_core' => 20
-    }
+  def dhcpd_conf_file?
+    !dhcpd_conf_file.nil?
```

Comment:
You could use `dhcp.yml` (from `settings.d`) to check if the dhcp module is enabled (`:enabled`). Then if `:use_provider` is `dhcp_isc` then the ISC config should be validated. We also have the `dhcp_remote_isc` plugin which reads `dhcp_remote_isc.yml` and is mostly similar to the `dhcp_isc` provider.